jcommune-service/src/main/java/org/jtalks/jcommune/service/transactional/TransactionalPostService.java

```diff
@@ -149,11 +150,14 @@ public void deletePost(Post post) {
             branch.clearLastPost();
         }
 
-
         // todo: event API?
         topicDao.saveOrUpdate(topic);
         securityService.deleteFromAcl(post);
-        notificationService.subscribedEntityChanged(topic);
+
+        Set<JCUser> topicSubscribers = new HashSet<>(topic.getSubscribers());
+        topicSubscribers.remove(user);
+        notificationService.subscribedEntityChanged(topic, topicSubscribers);
```

Comment:
I think that from the access to the database (subscribtion service) not to get rid of, because the creator of the post can be unsubscribed from notification. So send him a notice about the removal of his post, it makes no sense. Or are you suggesting to use the "topic.getSubscriberts()" method instead of "subscriptionService.getAllowedSubscribers()"?